lib/_stream_writable.js

```diff
@@ -199,6 +199,14 @@ WritableState.prototype.getBuffer = function getBuffer() {
   return out;
 };
 
+ObjectDefineProperties(WritableState.prototype, {
+  sync: {
+    get() {
+      return this.tick === process.tickId;
```

Comment:
@mcollina: one way to get around the `readable-stream` problem could be:

```js
process.tickId === undefined || this.tick === process.tickId
```

The downside would be that it would be slower (but probably correct?) on old node versions, i.e . it would always think it is "sync" and use `nextTick`.

Reply:
readable-stream has to work on old version of Node.js as well, as long as browsers. I would prefer to avoid this completely.